sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -165,9 +165,10 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
 
     case CreateTable(ResolvedDBObjectName(catalog, ident), schema, partitioning,
         tableSpec, ifNotExists) =>
-      val qualifiedLocation = tableSpec.location.map(makeQualifiedDBObjectPath(_))
+      val tableSpecWithQualifiedLocation = tableSpec.copy(
```

Comment:
shall we add a new method `qualifyLocInTableSpec(tableSpec): TableSpec` to simplify the code here and other places?